calico_containers/pycalico/util.py

```diff
@@ -10,3 +27,43 @@ def generate_cali_interface_name(prefix, ep_id):
     if len(prefix) > 4:
         raise ValueError('Prefix must be 4 characters or less.')
     return prefix + ep_id[:11]
+
+
+def get_host_ips(version=4, exclude=[]):
+    """
+    Gets all IP addresses assigned to this host.
+
+    Ignores Loopback Addresses
+
+    This function is fail-safe and will return an empty array instead of
+    raising any exceptions.
+
+    :param version: Desired version of IP addresses. Can be 4 or 6. defaults to 4
+    :param exclude: list of interfaces (strings) to ignore (ex. ["lo","docker0"])
+    :return: List of string representations of IP Addresses.
+    """
+    ip_addrs = []
+
+    # Select Regex for IPv6 or IPv4.
+    IP_RE = IPV4_RE if version is 4 else IPV6_RE
+
+    # Call `ip addr`.
+    try:
+        ip_addr_output = check_output(["ip", "-%d" % (version), "addr"])
+    except CalledProcessError:
+        print "Call to 'ip addr' Failed"
+        sys.exit(1)
+
+    # Separate interface blocks from ip addr output and iterate.
+    for iface_block in INTERFACE_SPLIT_RE.findall(ip_addr_output):
+        # Exclude certain interfaces.
+        match = IFACE_RE.match(iface_block)
+        if match and (match.group(1) not in exclude):
```

Comment:
PEP8 considers superfluous brackets to be an error. Should be highlighted in Pycharm? (LMK if you need more detail on this)